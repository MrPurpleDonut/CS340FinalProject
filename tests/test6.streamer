funk num addThem(num x, num y)
{
  num z = x + y;
  return z;
}

print addThem(40, 2);
