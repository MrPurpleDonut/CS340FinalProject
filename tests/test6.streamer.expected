Program parsed correctly!
---------------------------
--------Printing AST-------
---------------------------
Program
    Function List
        Function declaration: addThem
            Type: num
            Parameter List
                Parameter: x
                    Type: num
                Parameter: y
                    Type: num
            Statement List
                Declaration statement: z
                    Type: num
                    Assignment statement: z
                        Binary operator '+'
                            Identifier: x
                            Identifier: y
                Return Statement
                    Identifier: z
    Statement List
        Print Statement
            Function call: addThem
                Expression List
                    Number: 40.0
                    Number: 2.0
